test/Base_Tests/src/Semantic/Multi_Value_Spec.enso

```diff
@@ -14,6 +26,48 @@ add_specs suite_builder =
             (c:Complex).re . should_equal 1.5
             (c:Float) . should_equal 1.5
 
+    suite_builder.group "Chain Multi Value" group_builder->
+        to_b_to_c a =
+            b = a : B
+            c = b : C
+            c
+
+        group_builder.specify "Multiple conversions of A work" <|
+            a = A
+            a_text = to_b_to_c a
+            a_text . to_text . should_equal "(C (B A))"
+
+        group_builder.specify "Multiple conversions of A & X work" <|
+            ax = A:A&X
+            a_text = to_b_to_c ax
+            a_text . to_text . should_equal "(C (B A & X))"
+
+        group_builder.specify "Multiple conversions of A & X work" <|
+            xa = A:X&A
+            a_text = to_b_to_c xa
+            a_text . to_text . should_equal "(C (B A & X))"
+
+        group_builder.specify "Multiple conversions of (A & X : A) work" <|
+            ax = A:A&X
+            a = ax : A
+            a_text = to_b_to_c a
+            a_text . to_text . should_equal "(C (B A & X))"
+
+        group_builder.specify "Multiple conversions of X fail" <|
+            x = X
+
+            # has to fail as X cannot be converted to B in to_b_to_c
+            Test.expect_panic Type_Error <|
+                to_b_to_c x
+
+        group_builder.specify "Multiple conversions of (A & X : X) work" <|
+            ax = A:A&X
+            x = ax : X
+            # should it fail? X cannot be converted to B, but there is "extra"
+            # A in the x and allows the to_b_to_c conversion to succeed
```

Comment:
If we want this to fail, then we need to restrict the `from` conversions of `EnsoMultiValue` only to the `methodDispatchTypes`...